src/hera/workflows/_context.py

```diff
@@ -22,14 +20,12 @@ class SubNodeMixin(BaseMixin):
     """SubNodeMixin ensures that the class gets added to the Hera context on initialization.
 
     The mixin implements the core Hera `__hera_init__`, which is invoked post Hera object initialization. Anything
-    that inherits from this mixin has the capacity to manage a context via either being added to a context (like being
-    added to a Workflow/DAG) or managing a context itself (like holding Container, Script, etc).
+    that inherits from this mixin will add itself to the managed context (e.g. added to a Workflow/DAG).
     """
 
-    def __hera_init__(self: TNode) -> TNode:
+    def __hera_init__(self: "SubNodeMixin"):
```

Comment:
Does this work? Typing `self` isn't normally required, and not typing the return as `None` can cause issues.
```suggestion
    def __hera_init__(self) -> None:
```